lightning-invoice/src/lib.rs

```diff
@@ -1255,9 +1263,38 @@ impl Invoice {
 	///
 	/// (C-not exported) as we don't support Vec<&NonOpaqueType>
 	pub fn fallbacks(&self) -> Vec<&Fallback> {
+
+
 		self.signed_invoice.fallbacks()
 	}
 
+	/// Returns a list of all fallback addresses as [`Address`]es
+	pub fn fallback_addresses(&self) -> Vec<Address> {
+		self.fallbacks().iter().map(|fallback| {
+			let network = match self.currency() {
+				Currency::Bitcoin => Network::Bitcoin,
+				Currency::BitcoinTestnet => Network::Testnet,
+				Currency::Regtest => Network::Regtest,
+				Currency::Simnet => Network::Regtest,
+				Currency::Signet => Network::Signet,
+			};
+		
+			let script = match fallback {
+				Fallback::SegWitProgram { version, program } => {
+					Script::new_witness_program(*version, program.as_slice())
+				}
+				Fallback::PubKeyHash(pkh) => {
+					Script::new_p2pkh(&pkh)
+				}
+				Fallback::ScriptHash(sh) => {
+					Script::new_p2sh(&sh)
+				}
```

Comment:
No need to take a reference here since `pkh` and `sh` are already reference types.